components/datadog/apps/nginx/ecsFargateReadonly.go

```diff
@@ -0,0 +1,112 @@
+package nginx
+
+import (
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	fakeintakeComp "github.com/DataDog/test-infra-definitions/components/datadog/fakeintake"
+	ecsComp "github.com/DataDog/test-infra-definitions/components/ecs"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	ecsClient "github.com/DataDog/test-infra-definitions/resources/aws/ecs"
+	classicECS "github.com/pulumi/pulumi-aws/sdk/v6/go/aws/ecs"
+	"github.com/pulumi/pulumi-awsx/sdk/v2/go/awsx/ecs"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func FargateReadonlyAppDefinition(e aws.Environment, clusterArn pulumi.StringInput, apiKeySSMParamName pulumi.StringInput, fakeIntake *fakeintakeComp.Fakeintake, opts ...pulumi.ResourceOption) (*ecsComp.Workload, error) {
+	namer := e.Namer.WithPrefix("nginx").WithPrefix("fg-readonly")
+
+	opts = append(opts, e.WithProviders(config.ProviderAWS, config.ProviderAWSX))
+
+	EcsFargateComponent := &ecsComp.Workload{}
+	if err := e.Ctx().RegisterComponentResource("dd:apps", namer.ResourceName("grp"), EcsFargateComponent, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(EcsFargateComponent))
+
+	serverContainer := &ecs.TaskDefinitionContainerDefinitionArgs{
+		Name:  pulumi.String("nginx"),
+		Image: pulumi.String("ghcr.io/datadog/apps-nginx-server:main"),
+		DockerLabels: pulumi.StringMap{
+			"com.datadoghq.ad.checks": pulumi.String(utils.JSONMustMarshal(
+				map[string]interface{}{
+					"nginx": map[string]interface{}{
+						"init_config": map[string]interface{}{},
+						"instances": []map[string]interface{}{
+							{
+								"nginx_status_url": "http://%%host%%/nginx_status",
+							},
+						},
+					},
+				},
+			)),
+			"com.datadoghq.ad.tags": pulumi.String("[\"ecs_launch_type:fargate\"]"),
+		},
+		Cpu:       pulumi.IntPtr(100),
+		Memory:    pulumi.IntPtr(96),
+		Essential: pulumi.BoolPtr(true),
+		DependsOn: ecs.TaskDefinitionContainerDependencyArray{
+			ecs.TaskDefinitionContainerDependencyArgs{
+				ContainerName: pulumi.String("datadog-agent"),
+				Condition:     pulumi.String("HEALTHY"),
+			},
+		},
+		PortMappings: ecs.TaskDefinitionPortMappingArray{
+			ecs.TaskDefinitionPortMappingArgs{
+				ContainerPort: pulumi.IntPtr(80),
+				HostPort:      pulumi.IntPtr(80),
+				Protocol:      pulumi.StringPtr("tcp"),
+			},
+		},
+		HealthCheck: ecs.TaskDefinitionHealthCheckArgs{
+			Command: pulumi.StringArray{
+				pulumi.String("CMD-SHELL"),
+				pulumi.String("apk add curl && curl --fail http://localhost || exit 1"),
+			},
+		},
+		LogConfiguration: ecsClient.GetFirelensLogConfiguration(pulumi.String("nginx"), pulumi.String("nginx"), apiKeySSMParamName),
+	}
+
+	queryContainer := &ecs.TaskDefinitionContainerDefinitionArgs{
+		Name:  pulumi.String("query"),
+		Image: pulumi.String("ghcr.io/datadog/apps-http-client:main"),
```

Comment:
In order to get better test reproducibility, the use of floating `main` tag is now deprecated and should be replaced by “fixed” tags:

```suggestion
		Image: pulumi.String("ghcr.io/datadog/apps-http-client:" + apps.Version),
```

Reply:
Ahh thank you for the context! I branched off before that change was in place